### PR TITLE
bpo-18319: gettext() can retrieve a message even if a plural form exists

### PR DESCRIPTION
**Pull request by gbassiere**
Let's say you have this in your code:
```
gettext('egg')
```
and elsewhere this :
```
ngettext('egg', 'eggs', n)
```
GNU gettext tools extracts these as the same unique message and sensibly creates a pluralized entry in the PO file.

In the Python gettext module, when the MO file is parsed, a singular 'egg' will be stored as:
```
catalog['egg']
```
whereas a pluralized 'egg' is stored as:
```
catalog[('egg', 0)]
catalog[('eggs', 1)]
```
As a consequence, `gettext('egg')` fail to retrieve translations and you have to resort to `ngettext('egg', 'eggs', 1)` which is odd.

My suggestion is simply to modify `gettext()` so that is looks for `catalog[('egg', 0)]` if `catalog['egg']` didn't match.

https://bugs.python.org/issue18319

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@gbassiere

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Lib/gettext.py [hunk 1]
```diff
@@ -493,9 +493,13 @@ def gettext(self, message):
         missing = object()
         tmsg = self._catalog.get(message, missing)
         if tmsg is missing:
-            if self._fallback:
-                return self._fallback.gettext(message)
-            return message
+            try:
+                # if `message` has plural forms
+                tmsg = self._catalog[(message, self.plural(1))]
```
Comment (remilapeyre) [inline Lib/gettext.py]:
```suggestion
                return self._catalog[(message, self.plural(1))]
```

**Review by remilapeyre — APPROVED** [review]

File: Lib/gettext.py [hunk 2]
```diff
@@ -493,9 +493,13 @@ def gettext(self, message):
         missing = object()
         tmsg = self._catalog.get(message, missing)
         if tmsg is missing:
-            if self._fallback:
-                return self._fallback.gettext(message)
-            return message
+            try:
```
Comment (taleinat) [inline Lib/gettext.py]: I suggest simply adding the following before the existing `if tmsg is missing: ...` block:

```python3
        if tmsg is missing:
            tmsg = self._catalog.get((message, self.plural(1)), missing)
```

**Comment by ThiefMaster** [comment]
Just noticed this annoying bug as well (via babel). It's impossible to work around without starting to mess around with message contexts (and I should NOT need to add garbage to my code (and pot files) because of this), so it would be great if this fix could get merged..

**Comment by akulakov** [comment]
Should this logic be also added to `pgettext()` ?

**Pull request merged**